### ISSUE TITLE
Performance transactions doesn't contains context set in before_action

### ISSUE THREAD

**Issue by nicolas-brousse** [getsentry/sentry-ruby#1153]
**Describe the bug**
Performance transactions doesn't contains context set in a `before_action`.

**To Reproduce**
In `ApplicationController`

```rb
class ApplicationController < ActionController::Base
  before_action :set_sentry_contexts

  private

  def set_sentry_contexts
    Sentry.set_user(id: Current&.user&.id, ip_address: request.remote_ip)
    # Sentry.set_extras(params: params.to_unsafe_h, url: request.url)
    Sentry.set_tags(locale: I18n.locale)
  end
end
```

**Expected behavior**
Having user and locale information present in Transaction summary in sentry.

**Actual behavior**
Information are absent.

**Environment**
 - Ruby Version: 2.7.2
 - SDK Version: `sentry-ruby 4.0.1`
 - Integration Versions (if any):`sentry-rails 4.0.0` and `sentry-sidekiq 4.0.0`
 - Rails 6.0, Sidekiq 6.1.2

**Raven Config**

```ruby
Sentry.init do |config|
  config.traces_sample_rate = 0.5
  config.release = ENV["HEROKU_SLUG_COMMIT"]
  config.async = ->(event) { ErrorsTracker::SendEventJob.perform_later(event) }
  config.breadcrumbs_logger = %i[sentry_logger active_support_logger]
end
```

**Comment by st0012** [comment]
thanks for reporting, I've noticed this issue and have fixed it. the upcoming `4.1.0` version will contain the fix 😄

**Comment by nicolas-brousse** [comment]
@st0012 good to know! thanks for your answer and sorry for the noise, I search but didn't found something related to.

Also good job for the performance feature, looks awesome 🙂

**Comment by st0012** [comment]
no worries at all, I should've mentioned it in the PR 😬

**Comment by st0012** [comment]
closing because this has been released in `4.1.0` 🙂

**Comment by nicolas-brousse** [comment]
Thanks!
